### PR TITLE
Make Enchants showing as Scroll Tooltip.

### PR DESCRIPTION
**Pull request by tywtyw2002**
- Make Enchants showing as Scroll Tooltip.
- Enchanting Profit now is calc based on Scroll price.

### CODE REVIEW

**Comment by b-morgan** [comment]
Can you describe what you are trying to do with this pull request?

I'm not sure I need two sets of scroll data, you are adding to Skillet.EnchantSpellToItem when there is already a table Skillet.scrollData (at the end of the file). These two tables are separate and I believe they should stay that way.

Issue #100 had a pending fix to TradeSkillInfo.lua. I have committed that change and there is now a conflict.

**Comment by tywtyw2002** [comment]
The change for `Make Enchants showing as Scroll Tooltip` is to let enchant to show the scroll item tooltip instead of the enchant description and when `shift+click` will send the scroll item Link.
I think the conflict is he made `shift+click` send the enchant Link, and I made it send the scroll Item Link.

IMG_5540
I donot know why, in my laptop, the wow screenshot is full black.

The `EnchantSpellToItem` seems like used to calc the craft profit by plugin `Auctionator`, `Auctioneer`, `Auctipus`. When I carefully review the code, you are right, I should remove the data in `EnchantSpellToItem`, those are duplicate.

**Comment by b-morgan** [comment]
I have two reservations about this pull request.

First, I'm not sure that this change is suitable for the entire user base. I have released 1.49-alpha1 which contains a new option, enchant_scrolls (default false). Please incorporate this option (when true) in your changes. If you think the name and/or the description of this option needs changing, I'm open to your suggestions.

Second, I'm not comfortable with the wholescale replacement of the contents of the Skillet.scrollData table. Please convince me that this is necessary or just use the original table.

**Comment by tywtyw2002** [comment]
My scroll data are coming from wow DBC table (wow.tools) instead of Wowhead.

The reason I do this change is that the existed scroll data seems like not fully correct. 
For example.
`[158914] = 110638, -- Enchant Ring - Gift of Critical Strike` this did not exist in Wotlk.

**Comment by b-morgan** [comment]
I had a similar problem with Skillet (_retail_) so I wrote a little code to compare two tables from different sources. I've included a slightly modified version in the zip file below. `SkilletData2.lua` contains your scrollData table (renamed) and the code which can be run manually. `SkilletData.lua` contains the two items from your table that were missing from my table. I'll commit that in a bit.

My table is larger than yours because its source was from _retail_ and I just deleted the obvious stuff from future updates. Other than some extra entries, the data in the two tables is identical. Some of those extra entries might be from future WotLK releases or maybe Cataclysm Classic if it happens. Other than a small amount of wasted memory, it doesn't hurt to keep them.

[SkilletData2.zip](https://github.com/b-morgan/Skillet-Classic/files/9578990/SkilletData2.zip)

Bottom line, you can use my table!

**Review by b-morgan — APPROVED** [review]

**Comment by b-morgan** [comment]
Would you be willing to help me with an enhancement to Skillet-Classic? 

I need a table similar to the scrollData table that is indexed by the enchant recipeID (or the scrollID) with the value being the itemID of the appropriate Vellum, i.e. one of the six values:
```
		[38682] = 2, -- Armor Vellum
		[37602] = 2, -- Armor Vellum II
		[43145] = 2, -- Armor Vellum III
		[39349] = 2, -- Weapon Vellum
		[39350] = 2, -- Weapon Vellum II
		[43146] = 2, -- Weapon Vellum III
```
As I understand it, Armor/Weapon Vellum II is used if the enchant requires an item above level 35 and Armor/Weapon Vellum III is used if the enchant requires an item above level 60.

Skillet (retail) has a feature where you can right-Click (or alt-Click) on the Create, or Process buttons to apply the enchant to a vellum automagically. I'd like to implement this feature on Wrath Classic. Blizzard simplified the process in some expansion beyond Wrath to only have one kind of vellum. The Wrath Classic implementation needs to know the right vellum ID to apply each enchant to hence the need for this table (unless you have a better idea).

I don't want to generate this table manually as there's 200+ enchants. The table could just have entries for 5 of the 6 items (all but plain Armor Vellum) and I could default to using that one if an entry doesn't exist.
